third_party/move/move-model/src/builder/module_builder.rs

```diff
@@ -1222,10 +1265,15 @@ impl<'env, 'translator> ModuleBuilder<'env, 'translator> {
         let fields = match &def.fields {
             EA::StructFields::Defined(fields) => {
                 let mut field_map = BTreeMap::new();
+                let mut field_ty_and_locs = Vec::new(); // fix borrowing issues
```

Comment:
The commented code is the straightforward way to perform the check; I hope it helps reading the code. Anyway, it doesn't affect the compiled code.

Reply:
My main peeve was with the commented out code (that would not have worked), because that can lead to maintenance issues as the code evolves.

However, what you have changed it into (purely comments) is close to what I was asking: so we can consider this issue resolved.